### PR TITLE
feat: async for build step

### PR DESCRIPTION
**Pull request by tutevych**
## Purpose of this PR
add async for build step

## Testing status
No tests have been added.

### Manual testing status
Also, need to be tested with google-doc-connector

### CODE REVIEW

File: com.stansassets.build/Editor/Steps/IBuildStep.cs [hunk 1]
```diff
@@ -2,9 +2,10 @@
 {
     public interface IBuildStep
     {
-        bool Execute(BuildContext buildContext);
+        event Action<ExecuteFinishedArgs> OnExecuteFinished;
+
+        void Execute(BuildContext buildContext);
 
         int Priority { get; }
-        string GetResultMessage();
```
Comment (stan-osipov) [inline com.stansassets.build/Editor/Steps/IBuildStep.cs:8]: Where `ExecuteFinishedArgs` ? 

regarding the API the same comment how do I know as a developer when I need to call `OnExecuteFinished `
 void Execute(BuildContext buildContext, Action<ExecuteFinishedArgs> onComplete);

**Review by alexey-yaremenko — CHANGES_REQUESTED** [review]
This PR has compilation errors in Unity Editor. @tutevych , please, fix them
image

File: com.stansassets.build/Editor/Steps/ExecuteFinishedArgs.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace StansAssets.Build.Editor
+{
+    public class ExecuteFinishedArgs
```
Comment (alexey-yaremenko) [inline com.stansassets.build/Editor/Steps/ExecuteFinishedArgs.cs:7]: Some fields are missing in this class. How can I know if the build step has been finished successfully?
What about string ErrorMessage if step failed?

**Review by alexey-yaremenko — CHANGES_REQUESTED** [review]

File: com.stansassets.build/Editor/Steps/ExecuteFinishedArgs.cs (hunk 2 above)
Comment (stan-osipov) [reply com.stansassets.build/Editor/Steps/ExecuteFinishedArgs.cs:7]: I don't really like this mode. Let's talk more about it

File: com.stansassets.build/Editor/Steps/IBuildStep.cs [hunk 3]
```diff
@@ -1,10 +1,13 @@
...
+﻿using System;
+
+namespace StansAssets.Build.Editor
 {
     public interface IBuildStep
     {
-        bool Execute(BuildContext buildContext);
+        event Action<ExecuteFinishedArgs> OnExecuteFinished;
+
+        void Execute(BuildContext buildContext, Action onComplete = null);
```
Comment (stan-osipov) [inline com.stansassets.build/Editor/Steps/IBuildStep.cs:9]: Sorry, why do we need both?

File: com.stansassets.build/Editor/Steps/UnityPlayerBuildStep.cs [hunk 4]
```diff
@@ -12,27 +13,27 @@ public class UnityPlayerBuildStep : IBuildStep
...
-        private bool BuildProject()
+        private void BuildProject()
         {
-            bool isSuccess = true;
-
             BuildPlayerOptions defaultBuildPlayerOptions = new BuildPlayerOptions();
             BuildPlayerOptions currentBuildPlayerOptions =
                 BuildPlayerWindow.DefaultBuildMethods.GetBuildPlayerOptions(defaultBuildPlayerOptions);
 
+            ExecuteFinishedArgs executeFinishedArgs = new ExecuteFinishedArgs();
```
Comment (stan-osipov) [inline com.stansassets.build/Editor/Steps/UnityPlayerBuildStep.cs:31]: Talke about that in the Nazar's PR

Comment (stan-osipov) [reply com.stansassets.build/Editor/Steps/UnityPlayerBuildStep.cs:31]: We can leave it as is just for the sake for merging this and add tasks so we won't forget to change it

**Review by stan-osipov — CHANGES_REQUESTED** [review]

**Comment by PavelKlym** [comment]
Closing this PR because Nazar worked on requested changes in [another PR](https://github.com/StansAssets/com.stansassets.build/pull/3).

**Pull request closed without merging**